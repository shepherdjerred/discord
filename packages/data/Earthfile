VERSION 0.8

deno-node:
  ARG NODE_MAJOR=20
  FROM ../../+deno
  RUN apt update
  RUN apt install -y ca-certificates curl gnupg
  RUN mkdir -p /etc/apt/keyrings
  RUN curl -fsSL https://deb.nodesource.com/gpgkey/nodesource-repo.gpg.key | gpg --dearmor -o /etc/apt/keyrings/nodesource.gpg
  RUN echo "deb [signed-by=/etc/apt/keyrings/nodesource.gpg] https://deb.nodesource.com/node_$NODE_MAJOR.x nodistro main" | tee /etc/apt/sources.list.d/nodesource.list
  RUN apt update
  RUN apt install nodejs -y
  RUN npm install -g npm

build:
  FROM +deno-node
  COPY . packages/data/
  WORKDIR packages/data
  RUN deno task build
  SAVE ARTIFACT dist

src:
  FROM +deno-node
  COPY . .
  SAVE ARTIFACT *

check:
  FROM +src
  RUN deno check src/mod.ts
  RUN deno lint
