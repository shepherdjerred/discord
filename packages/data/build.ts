import { build, emptyDir } from "https://deno.land/x/dnt/mod.ts";

await emptyDir("./dist");

await build({
  entryPoints: ["./src/index.ts"],
  outDir: "./dist",
  test: false,
  shims: {
    deno: true,
  },
  package: {
    name: "@discord/data",
    version: Deno.args[0],
  },
});
