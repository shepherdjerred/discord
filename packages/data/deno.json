{
  "tasks": {
    "check": "deno check src/index.ts",
    "check:watch": "deno run -A https://deno.land/x/denon/denon.ts check",
    "build": "deno run -A build.ts",
    "test:update": "deno test -A --unstable-ffi -- --update"
  },
  "lint": {
    "exclude": ["dist"]
  }
}
