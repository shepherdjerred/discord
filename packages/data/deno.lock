{
  "version": "3",
  "packages": {
    "specifiers": {
      "npm:@types/lodash": "npm:@types/lodash@4.14.202",
      "npm:lodash@4.17.21": "npm:lodash@4.17.21",
      "npm:twisted@1.57.0": "npm:twisted@1.57.0"
    },
    "npm": {
      "@types/lodash@4.14.202": {
        "integrity": "sha512-OvlIYQK9tNneDlS0VN54LLd5uiPCBOp7gS5Z0f1mjoJYBrtStzgmJBxONW3U6OZqdtNzZPmn9BS/7WI7BFFcFQ==",
        "dependencies": {}
      },
      "asynckit@0.4.0": {
        "integrity": "sha512-Oei9OH4tRh0YqU3GxhX79dM/mwVgvbZJaSNaRk+bshkj0S5cfHcgYakreBjrHwatXKbz+IoIdYLxrKim2MjW0Q==",
        "dependencies": {}
      },
      "axios@1.6.7": {
        "integrity": "sha512-/hDJGff6/c7u0hDkvkGxR/oy6CbCs8ziCsC7SqmhjfozqiJGc8Z11wrv9z9lYfY4K8l+H9TpjcMDX0xOZmx+RA==",
        "dependencies": {
          "follow-redirects": "follow-redirects@1.15.5",
          "form-data": "form-data@4.0.0",
          "proxy-from-env": "proxy-from-env@1.1.0"
        }
      },
      "combined-stream@1.0.8": {
        "integrity": "sha512-FQN4MRfuJeHf7cBbBMJFXhKSDq+2kAArBlmRBvcvFE5BB1HZKXtSFASDhdlz9zOYwxh8lDdnvmMOe/+5cdoEdg==",
        "dependencies": {
          "delayed-stream": "delayed-stream@1.0.0"
        }
      },
      "delayed-stream@1.0.0": {
        "integrity": "sha512-ZySD7Nf91aLB0RxL4KGrKHBXl7Eds1DAmEdcoVawXnLD7SDhpNgtuII2aAkg7a7QS41jxPSZ17p4VdGnMHk3MQ==",
        "dependencies": {}
      },
      "dotenv@16.4.5": {
        "integrity": "sha512-ZmdL2rui+eB2YwhsWzjInR8LldtZHGDoQ1ugH85ppHKwpUHL7j7rN0Ti9NCnGiQbhaZ11FpR+7ao1dNsmduNUg==",
        "dependencies": {}
      },
      "follow-redirects@1.15.5": {
        "integrity": "sha512-vSFWUON1B+yAw1VN4xMfxgn5fTUiaOzAJCKBwIIgT/+7CuGy9+r+5gITvP62j3RmaD5Ph65UaERdOSRGUzZtgw==",
        "dependencies": {}
      },
      "form-data@4.0.0": {
        "integrity": "sha512-ETEklSGi5t0QMZuiXoA/Q6vcnxcLQP5vdugSpuAyi6SVGi2clPPp+xgEhuMaHC+zGgn31Kd235W35f7Hykkaww==",
        "dependencies": {
          "asynckit": "asynckit@0.4.0",
          "combined-stream": "combined-stream@1.0.8",
          "mime-types": "mime-types@2.1.35"
        }
      },
      "http-status-codes@2.3.0": {
        "integrity": "sha512-RJ8XvFvpPM/Dmc5SV+dC4y5PCeOhT3x1Hq0NU3rjGeg5a/CqlhZ7uudknPwZFz4aeAXDcbAyaeP7GAo9lvngtA==",
        "dependencies": {}
      },
      "lodash@4.17.21": {
        "integrity": "sha512-v2kDEe57lecTulaDIuNTPy3Ry4gLGJ6Z1O3vE1krgXZNrsQ+LFTGHVxVjcXPs17LhbZVGedAJv8XZ1tvj5FvSg==",
        "dependencies": {}
      },
      "mime-db@1.52.0": {
        "integrity": "sha512-sPU4uV7dYlvtWJxwwxHD0PuihVNiE7TyAbQ5SWxDCB9mUYvOgroQOwYQQOKPJ8CIbE+1ETVlOoK1UC2nU3gYvg==",
        "dependencies": {}
      },
      "mime-types@2.1.35": {
        "integrity": "sha512-ZDY+bPm5zTTF+YpCrAU9nK0UgICYPT0QtT1NZWFv4s++TNkcgVaT0g6+4R2uI4MjQjzysHB1zxuWL50hzaeXiw==",
        "dependencies": {
          "mime-db": "mime-db@1.52.0"
        }
      },
      "promise-queue@2.2.5": {
        "integrity": "sha512-p/iXrPSVfnqPft24ZdNNLECw/UrtLTpT3jpAAMzl/o5/rDsGCPo3/CQS2611flL6LkoEJ3oQZw7C8Q80ZISXRQ==",
        "dependencies": {}
      },
      "proxy-from-env@1.1.0": {
        "integrity": "sha512-D+zkORCbA9f1tdWRK0RaCR3GPv50cMxcrz4X8k5LTSUD1Dkw47mKJEZQNunItRTkWwgtaUSo1RVFRIG9ZXiFYg==",
        "dependencies": {}
      },
      "twisted@1.57.0": {
        "integrity": "sha512-5NeGxtxalP+IyFqTZ0zMWNk6Q4eCH8GiQEnVo6ofTE4fctKoTHW76zIkjBEYtyPnX1Dxjn55rSXMAHilw2teMw==",
        "dependencies": {
          "axios": "axios@1.6.7",
          "dotenv": "dotenv@16.4.5",
          "http-status-codes": "http-status-codes@2.3.0",
          "lodash": "lodash@4.17.21",
          "promise-queue": "promise-queue@2.2.5",
          "uuid": "uuid@9.0.1"
        }
      },
      "uuid@9.0.1": {
        "integrity": "sha512-b+1eJOlsR9K8HJpow9Ok3fiWOWSIcIzXodvv0rQjVoOVNpWMpxf1wZNpt4y9h10odCNrqnYp1OBzRktckBe3sA==",
        "dependencies": {}
      }
    }
  },
  "redirects": {
    "https://deno.land/x/dnt/mod.ts": "https://deno.land/x/dnt@0.39.0/mod.ts"
  },
  "remote": {
    "https://deno.land/std@0.140.0/_util/assert.ts": "e94f2eb37cebd7f199952e242c77654e43333c1ac4c5c700e929ea3aa5489f74",
    "https://deno.land/std@0.140.0/_util/os.ts": "3b4c6e27febd119d36a416d7a97bd3b0251b77c88942c8f16ee5953ea13e2e49",
    "https://deno.land/std@0.140.0/bytes/bytes_list.ts": "67eb118e0b7891d2f389dad4add35856f4ad5faab46318ff99653456c23b025d",
    "https://deno.land/std@0.140.0/bytes/equals.ts": "fc16dff2090cced02497f16483de123dfa91e591029f985029193dfaa9d894c9",
    "https://deno.land/std@0.140.0/bytes/mod.ts": "763f97d33051cc3f28af1a688dfe2830841192a9fea0cbaa55f927b49d49d0bf",
    "https://deno.land/std@0.140.0/fmt/colors.ts": "30455035d6d728394781c10755351742dd731e3db6771b1843f9b9e490104d37",
    "https://deno.land/std@0.140.0/fs/_util.ts": "0fb24eb4bfebc2c194fb1afdb42b9c3dda12e368f43e8f2321f84fc77d42cb0f",
    "https://deno.land/std@0.140.0/fs/ensure_dir.ts": "9dc109c27df4098b9fc12d949612ae5c9c7169507660dcf9ad90631833209d9d",
    "https://deno.land/std@0.140.0/io/buffer.ts": "bd0c4bf53db4b4be916ca5963e454bddfd3fcd45039041ea161dbf826817822b",
    "https://deno.land/std@0.140.0/path/_constants.ts": "df1db3ffa6dd6d1252cc9617e5d72165cd2483df90e93833e13580687b6083c3",
    "https://deno.land/std@0.140.0/path/_interface.ts": "ee3b431a336b80cf445441109d089b70d87d5e248f4f90ff906820889ecf8d09",
    "https://deno.land/std@0.140.0/path/_util.ts": "c1e9686d0164e29f7d880b2158971d805b6e0efc3110d0b3e24e4b8af2190d2b",
    "https://deno.land/std@0.140.0/path/common.ts": "bee563630abd2d97f99d83c96c2fa0cca7cee103e8cb4e7699ec4d5db7bd2633",
    "https://deno.land/std@0.140.0/path/glob.ts": "cb5255638de1048973c3e69e420c77dc04f75755524cb3b2e160fe9277d939ee",
    "https://deno.land/std@0.140.0/path/mod.ts": "d3e68d0abb393fb0bf94a6d07c46ec31dc755b544b13144dee931d8d5f06a52d",
    "https://deno.land/std@0.140.0/path/posix.ts": "293cdaec3ecccec0a9cc2b534302dfe308adb6f10861fa183275d6695faace44",
    "https://deno.land/std@0.140.0/path/separator.ts": "fe1816cb765a8068afb3e8f13ad272351c85cbc739af56dacfc7d93d710fe0f9",
    "https://deno.land/std@0.140.0/path/win32.ts": "31811536855e19ba37a999cd8d1b62078235548d67902ece4aa6b814596dd757",
    "https://deno.land/std@0.140.0/streams/conversion.ts": "712585bfa0172a97fb68dd46e784ae8ad59d11b88079d6a4ab098ff42e697d21",
    "https://deno.land/std@0.181.0/_util/asserts.ts": "178dfc49a464aee693a7e285567b3d0b555dc805ff490505a8aae34f9cfb1462",
    "https://deno.land/std@0.181.0/_util/os.ts": "d932f56d41e4f6a6093d56044e29ce637f8dcc43c5a90af43504a889cf1775e3",
    "https://deno.land/std@0.181.0/fmt/colors.ts": "d67e3cd9f472535241a8e410d33423980bec45047e343577554d3356e1f0ef4e",
    "https://deno.land/std@0.181.0/fs/_util.ts": "65381f341af1ff7f40198cee15c20f59951ac26e51ddc651c5293e24f9ce6f32",
    "https://deno.land/std@0.181.0/fs/empty_dir.ts": "c3d2da4c7352fab1cf144a1ecfef58090769e8af633678e0f3fabaef98594688",
    "https://deno.land/std@0.181.0/fs/ensure_dir.ts": "dc64c4c75c64721d4e3fb681f1382f803ff3d2868f08563ff923fdd20d071c40",
    "https://deno.land/std@0.181.0/fs/expand_glob.ts": "e4f56259a0a70fe23f05215b00de3ac5e6ba46646ab2a06ebbe9b010f81c972a",
    "https://deno.land/std@0.181.0/fs/walk.ts": "ea95ffa6500c1eda6b365be488c056edc7c883a1db41ef46ec3bf057b1c0fe32",
    "https://deno.land/std@0.181.0/path/_constants.ts": "e49961f6f4f48039c0dfed3c3f93e963ca3d92791c9d478ac5b43183413136e0",
    "https://deno.land/std@0.181.0/path/_interface.ts": "6471159dfbbc357e03882c2266d21ef9afdb1e4aa771b0545e90db58a0ba314b",
    "https://deno.land/std@0.181.0/path/_util.ts": "d7abb1e0dea065f427b89156e28cdeb32b045870acdf865833ba808a73b576d0",
    "https://deno.land/std@0.181.0/path/common.ts": "ee7505ab01fd22de3963b64e46cff31f40de34f9f8de1fff6a1bd2fe79380000",
    "https://deno.land/std@0.181.0/path/glob.ts": "d479e0a695621c94d3fd7fe7abd4f9499caf32a8de13f25073451c6ef420a4e1",
    "https://deno.land/std@0.181.0/path/mod.ts": "bf718f19a4fdd545aee1b06409ca0805bd1b68ecf876605ce632e932fe54510c",
    "https://deno.land/std@0.181.0/path/posix.ts": "8b7c67ac338714b30c816079303d0285dd24af6b284f7ad63da5b27372a2c94d",
    "https://deno.land/std@0.181.0/path/separator.ts": "0fb679739d0d1d7bf45b68dacfb4ec7563597a902edbaf3c59b50d5bcadd93b1",
    "https://deno.land/std@0.181.0/path/win32.ts": "d186344e5583bcbf8b18af416d13d82b35a317116e6460a5a3953508c3de5bba",
    "https://deno.land/x/code_block_writer@12.0.0/mod.ts": "2c3448060e47c9d08604c8f40dee34343f553f33edcdfebbf648442be33205e5",
    "https://deno.land/x/code_block_writer@12.0.0/utils/string_utils.ts": "60cb4ec8bd335bf241ef785ccec51e809d576ff8e8d29da43d2273b69ce2a6ff",
    "https://deno.land/x/deno_cache@0.6.2/auth_tokens.ts": "5d1d56474c54a9d152e44d43ea17c2e6a398dd1e9682c69811a313567c01ee1e",
    "https://deno.land/x/deno_cache@0.6.2/cache.ts": "58b53c128b742757efcad10af9a3871f23b4e200674cb5b0ddf61164fb9b2fe7",
    "https://deno.land/x/deno_cache@0.6.2/deno_dir.ts": "1ea355b8ba11c630d076b222b197cfc937dd81e5a4a260938997da99e8ff93a0",
    "https://deno.land/x/deno_cache@0.6.2/deps.ts": "12cca94516cf2d3ed42fccd4b721ecd8060679253f077d83057511045b0081aa",
    "https://deno.land/x/deno_cache@0.6.2/dirs.ts": "009c6f54e0b610914d6ce9f72f6f6ccfffd2d47a79a19061e0a9eb4253836069",
    "https://deno.land/x/deno_cache@0.6.2/disk_cache.ts": "66a1e604a8d564b6dd0500326cac33d08b561d331036bf7272def80f2f7952aa",
    "https://deno.land/x/deno_cache@0.6.2/file_fetcher.ts": "4f3e4a2c78a5ca1e4812099e5083f815a8525ab20d389b560b3517f6b1161dd6",
    "https://deno.land/x/deno_cache@0.6.2/http_cache.ts": "407135eaf2802809ed373c230d57da7ef8dff923c4abf205410b9b99886491fd",
    "https://deno.land/x/deno_cache@0.6.2/lib/deno_cache_dir.generated.js": "59f8defac32e8ebf2a30f7bc77e9d88f0e60098463fb1b75e00b9791a4bbd733",
    "https://deno.land/x/deno_cache@0.6.2/lib/snippets/deno_cache_dir-a2aecaa9536c9402/fs.js": "cbe3a976ed63c72c7cb34ef845c27013033a3b11f9d8d3e2c4aa5dda2c0c7af6",
    "https://deno.land/x/deno_cache@0.6.2/mod.ts": "b4004287e1c6123d7f07fe9b5b3e94ce6d990c4102949a89c527c68b19627867",
    "https://deno.land/x/deno_cache@0.6.2/util.ts": "f3f5a0cfc60051f09162942fb0ee87a0e27b11a12aec4c22076e3006be4cc1e2",
    "https://deno.land/x/dir@1.5.1/data_local_dir/mod.ts": "91eb1c4bfadfbeda30171007bac6d85aadacd43224a5ed721bbe56bc64e9eb66",
    "https://deno.land/x/dnt@0.39.0/lib/compiler.ts": "7f4447531581896348b8a379ab94730856b42ae50d99043f2468328360293cb1",
    "https://deno.land/x/dnt@0.39.0/lib/compiler_transforms.ts": "f21aba052f5dcf0b0595c734450842855c7f572e96165d3d34f8fed2fc1f7ba1",
    "https://deno.land/x/dnt@0.39.0/lib/mod.deps.ts": "8d6123c8e1162037e58aa8126686a03d1e2cffb250a8757bf715f80242097597",
    "https://deno.land/x/dnt@0.39.0/lib/npm_ignore.ts": "57fbb7e7b935417d225eec586c6aa240288905eb095847d3f6a88e290209df4e",
    "https://deno.land/x/dnt@0.39.0/lib/package_json.ts": "607b0a4f44acad071a4c8533b312a27d6671eac8e6a23625c8350ce29eadb2ba",
    "https://deno.land/x/dnt@0.39.0/lib/pkg/dnt_wasm.generated.js": "4f9c59b3ca6c875adabb10df256e273fff1129fca3a1557eb8936bddd7da7b18",
    "https://deno.land/x/dnt@0.39.0/lib/pkg/snippets/dnt-wasm-a15ef721fa5290c5/helpers.js": "aba69a019a6da6f084898a6c7b903b8b583bc0dbd82bfb338449cf0b5bce58fd",
    "https://deno.land/x/dnt@0.39.0/lib/shims.ts": "60fd285ad433c6944544595e7b885eab3eab09253252891380654f4cd3addaaa",
    "https://deno.land/x/dnt@0.39.0/lib/test_runner/get_test_runner_code.ts": "4dc7a73a13b027341c0688df2b29a4ef102f287c126f134c33f69f0339b46968",
    "https://deno.land/x/dnt@0.39.0/lib/test_runner/test_runner.ts": "4d0da0500ec427d5f390d9a8d42fb882fbeccc92c92d66b6f2e758606dbd40e6",
    "https://deno.land/x/dnt@0.39.0/lib/transform.deps.ts": "2e159661e1c5c650de9a573babe0e319349fe493105157307ec2ad2f6a52c94e",
    "https://deno.land/x/dnt@0.39.0/lib/types.ts": "b8e228b2fac44c2ae902fbb73b1689f6ab889915bd66486c8a85c0c24255f5fb",
    "https://deno.land/x/dnt@0.39.0/lib/utils.ts": "224f15f33e7226a2fd991e438d0291d7ed8c7889807efa2e1ecb67d2d1db6720",
    "https://deno.land/x/dnt@0.39.0/mod.ts": "9df36a862161d9eb376472b699f6cb08ba0ad1704e0826fbe13be766bd3c01da",
    "https://deno.land/x/dnt@0.39.0/transform.ts": "f68743a14cf9bf53bfc9c81073871d69d447a7f9e3453e0447ca2fb78926bb1d",
    "https://deno.land/x/ts_morph@20.0.0/bootstrap/mod.ts": "b53aad517f106c4079971fcd4a81ab79fadc40b50061a3ab2b741a09119d51e9",
    "https://deno.land/x/ts_morph@20.0.0/bootstrap/ts_morph_bootstrap.js": "6645ac03c5e6687dfa8c78109dc5df0250b811ecb3aea2d97c504c35e8401c06",
    "https://deno.land/x/ts_morph@20.0.0/common/DenoRuntime.ts": "6a7180f0c6e90dcf23ccffc86aa8271c20b1c4f34c570588d08a45880b7e172d",
    "https://deno.land/x/ts_morph@20.0.0/common/mod.ts": "01985d2ee7da8d1caee318a9d07664774fbee4e31602bc2bb6bb62c3489555ed",
    "https://deno.land/x/ts_morph@20.0.0/common/ts_morph_common.js": "2325f94f61dc5f3f98a1dab366dc93048d11b1433d718b10cfc6ee5a1cfebe8f",
    "https://deno.land/x/ts_morph@20.0.0/common/typescript.js": "b9edf0a451685d13e0467a7ed4351d112b74bd1e256b915a2b941054e31c1736",
    "https://deno.land/x/wasmbuild@0.15.1/cache.ts": "9d01b5cb24e7f2a942bbd8d14b093751fa690a6cde8e21709ddc97667e6669ed",
    "https://deno.land/x/wasmbuild@0.15.1/loader.ts": "8c2fc10e21678e42f84c5135d8ab6ab7dc92424c3f05d2354896a29ccfd02a63",
    "https://esm.sh/ts-pattern@5.0.5": "248bd637724edc10339cac795a8f584b47b06f2f338a7da356202a90023eab46",
    "https://esm.sh/v135/ts-pattern@5.0.5/denonext/ts-pattern.mjs": "cb88b9b005fdd512f6b988e99e3271f345c8fdfff0a4c65db6c56ae597aa6289",
    "https://esm.sh/v135/zod@3.22.4/denonext/zod.mjs": "660128af5d1e921745c4d452472d103d9f2fc5afa508bbf233b83d35a272ad67",
    "https://esm.sh/zod@3.22.4": "f7bfaf7e78bc8fa39e67e07c0dacd651925e889f5a4ef02b3ef6bab3925a8384"
  }
}
