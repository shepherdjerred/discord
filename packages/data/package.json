{
  "name": "@glitter-boys/data",
  "version": "1.0.0",
  "private": true,
  "main": "dist/index.js",
  "types": "dist/index.d.ts",
  "scripts": {
    "build": "tsc",
    "eslint:check": "eslint .",
    "prettier:check": "prettier --check .",
    "lint": "npm run lint:check",
    "lint:check": "npm run eslint:check && npm run prettier:check",
    "prettier:fix": "prettier --write ."
  },
  "type": "module",
  "dependencies": {
    "reflect-metadata": "^0.1.13",
    "typeorm": "https://github.com/shepherdjerred/typeorm"
  },
  "devDependencies": {
    "@types/node": "^20.2.5"
  }
}
