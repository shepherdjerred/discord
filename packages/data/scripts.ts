import type { DenonConfig } from "https://deno.land/x/denon@2.5.0/mod.ts";

export const config: DenonConfig = {
  scripts: {
    check: {
      cmd: "deno task check",
    },
  },
  watcher: {
    interval: 1000,
  },
};

export default config;
