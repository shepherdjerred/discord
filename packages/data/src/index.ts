export * from "./karma/index.js";
export * from "./model/index.js";
export * from "./image/index.js";
export * from "./dataDragon/index.js";
