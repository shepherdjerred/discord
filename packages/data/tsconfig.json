{
  "extends": "@tsconfig/strictest/tsconfig.json",
  "include": ["src/**/*"],
  "compilerOptions": {
    "module": "NodeNext",
    "outDir": "dist",
    "emitDecoratorMetadata": true,
    "experimentalDecorators": true,
    "declaration": true,
    "declarationMap": true,
    "jsx": "react"
  }
}
