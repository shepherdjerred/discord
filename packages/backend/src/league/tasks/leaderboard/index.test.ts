import { Division, Player, Rank, Tier } from "@glitter-boys/data";
import { describe, it, expect } from "vitest";
import { toLeaderboard } from "./index.js";
import _ from "lodash";

const createRank = ({ division, tier, lp }: { division: Division; tier: Tier; lp: number }): Rank => ({
  division,
  tier,
  lp,
  wins: 10,
  losses: 10,
});

const createPlayer = ({ name, currentRank }: { name: string; currentRank: Rank }): Player => ({
  config: {
    name,
    league: {
      leagueAccount: {
        id: "test",
        puuid: "test",
        accountId: "test",
      },
    },
    discordAccount: {
      id: "test",
    },
  },
  ranks: {
    solo: currentRank,
    flex: currentRank,
  },
});

describe("Leaderboard", () => {
  it("should return a correct leaderboard", () => {
    const players: Player[] = _.shuffle([
      createPlayer({
        name: "top of the leaderboard",
        currentRank: createRank({ division: 1, tier: "challenger", lp: 100 }),
      }),
      createPlayer({
        name: "bottom of the leaderboard",
        currentRank: createRank({ division: 4, tier: "iron", lp: 0 }),
      }),
      createPlayer({
        name: "positive",
        currentRank: createRank({ division: 3, tier: "silver", lp: 50 }),
      }),
      createPlayer({
        name: "negative",
        currentRank: createRank({ division: 2, tier: "bronze", lp: 50 }),
      }),
      createPlayer({
        name: "no change",
        currentRank: createRank({ division: 3, tier: "silver", lp: 50 }),
      }),
      createPlayer({
        name: "tied with next player",
        currentRank: createRank({ division: 1, tier: "silver", lp: 50 }),
      }),
      createPlayer({
        name: "tied with previous player",
        currentRank: createRank({ division: 3, tier: "silver", lp: 50 }),
      }),
    ]);

    expect(toLeaderboard(players)).toMatchSnapshot({
      date: expect.any(Date) as unknown,
    });
  });
});
