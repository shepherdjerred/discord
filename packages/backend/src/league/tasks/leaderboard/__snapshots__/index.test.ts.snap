export const snapshot = {};

snapshot[`should return a correct leaderboard 1`] = `
{
  contents: [
    {
      leaguePoints: 4000,
      player: {
        config: {
          discordAccount: {
            id: "test",
          },
          league: {
            leagueAccount: {
              accountId: "test",
              id: "test",
              puuid: "test",
              region: "AMERICA_NORTH",
            },
          },
          name: "top of the leaderboard",
        },
        ranks: {
          flex: {
            division: 1,
            losses: 10,
            lp: 100,
            tier: "challenger",
            wins: 10,
          },
          solo: {
            division: 1,
            losses: 10,
            lp: 100,
            tier: "challenger",
            wins: 10,
          },
        },
      },
      position: 1,
    },
    {
      leaguePoints: 1150,
      player: {
        config: {
          discordAccount: {
            id: "test",
          },
          league: {
            leagueAccount: {
              accountId: "test",
              id: "test",
              puuid: "test",
              region: "AMERICA_NORTH",
            },
          },
          name: "tied with next player",
        },
        ranks: {
          flex: {
            division: 1,
            losses: 10,
            lp: 50,
            tier: "silver",
            wins: 10,
          },
          solo: {
            division: 1,
            losses: 10,
            lp: 50,
            tier: "silver",
            wins: 10,
          },
        },
      },
      position: 2,
    },
    {
      leaguePoints: 950,
      player: {
        config: {
          discordAccount: {
            id: "test",
          },
          league: {
            leagueAccount: {
              accountId: "test",
              id: "test",
              puuid: "test",
              region: "AMERICA_NORTH",
            },
          },
          name: "no change",
        },
        ranks: {
          flex: {
            division: 3,
            losses: 10,
            lp: 50,
            tier: "silver",
            wins: 10,
          },
          solo: {
            division: 3,
            losses: 10,
            lp: 50,
            tier: "silver",
            wins: 10,
          },
        },
      },
      position: 3,
    },
    {
      leaguePoints: 950,
      player: {
        config: {
          discordAccount: {
            id: "test",
          },
          league: {
            leagueAccount: {
              accountId: "test",
              id: "test",
              puuid: "test",
              region: "AMERICA_NORTH",
            },
          },
          name: "tied with previous player",
        },
        ranks: {
          flex: {
            division: 3,
            losses: 10,
            lp: 50,
            tier: "silver",
            wins: 10,
          },
          solo: {
            division: 3,
            losses: 10,
            lp: 50,
            tier: "silver",
            wins: 10,
          },
        },
      },
      position: 4,
    },
    {
      leaguePoints: 950,
      player: {
        config: {
          discordAccount: {
            id: "test",
          },
          league: {
            leagueAccount: {
              accountId: "test",
              id: "test",
              puuid: "test",
              region: "AMERICA_NORTH",
            },
          },
          name: "positive",
        },
        ranks: {
          flex: {
            division: 3,
            losses: 10,
            lp: 50,
            tier: "silver",
            wins: 10,
          },
          solo: {
            division: 3,
            losses: 10,
            lp: 50,
            tier: "silver",
            wins: 10,
          },
        },
      },
      position: 5,
    },
    {
      leaguePoints: 650,
      player: {
        config: {
          discordAccount: {
            id: "test",
          },
          league: {
            leagueAccount: {
              accountId: "test",
              id: "test",
              puuid: "test",
              region: "AMERICA_NORTH",
            },
          },
          name: "negative",
        },
        ranks: {
          flex: {
            division: 2,
            losses: 10,
            lp: 50,
            tier: "bronze",
            wins: 10,
          },
          solo: {
            division: 2,
            losses: 10,
            lp: 50,
            tier: "bronze",
            wins: 10,
          },
        },
      },
      position: 6,
    },
    {
      leaguePoints: 0,
      player: {
        config: {
          discordAccount: {
            id: "test",
          },
          league: {
            leagueAccount: {
              accountId: "test",
              id: "test",
              puuid: "test",
              region: "AMERICA_NORTH",
            },
          },
          name: "bottom of the leaderboard",
        },
        ranks: {
          flex: {
            division: 4,
            losses: 10,
            lp: 0,
            tier: "iron",
            wins: 10,
          },
          solo: {
            division: 4,
            losses: 10,
            lp: 0,
            tier: "iron",
            wins: 10,
          },
        },
      },
      position: 7,
    },
  ],
  date: 2024-03-04T00:29:35.641Z,
}
`;
